### PR TITLE
Sockets Tatiana

### PR DESCRIPTION
**Pull request by tatsqui**
# JS Adagrams
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
--- | ---
What patterns were you able to use from your Ruby knowledge to apply to JavaScript? | Using Loops in a similar way to get what I needed. 
Did you need to use different strategies to find information online about JavaScript than you would for Ruby? | Yes, a lot of the info online is OLD, and the libraries have been updated. 
What was a challenge you were able to overcome on this assignment? | Wave 2 was a bit challenging. I also think figuring out the incremental steps to work out a tie was a bit challenging to figure out in a new language. The tests honestly were very helpful because they used some logic that I was able to harness in my code logic. 
What has been interesting and positive about learning a new programming language? | I think that it's cool to learn a language that is 'weakly typed' but feels more strongly typed than Ruby. 
What is something to focus on your learnings in JavaScript in the next week? | GEtting used to treating functions as objects.

### CODE REVIEW

File: src/adagrams.js [hunk 1]
```diff
@@ -1,8 +1,101 @@
 const Adagrams = {
   drawLetters() {
-    // Implement this method for wave 1
+    const allLetters = { A: 9, B: 2, C:2, D:4, E:12, F:2, 
+      G:3, H:2, I:9, J: 1, K:1, L:4, M: 2, N:6, O:8, P:2, Q:1, R:6, S:4,
+      T:6, U:4, V:2, W:2, X:1, Y:2, Z:1
+    }
```
Comment (droberts-sea) [inline src/adagrams.js:6]: I really like that you store the letter frequencies and then build the pool on the fly - this is much easier to read than a giant array full of 9 'A's, 2 'B's, etc.

File: src/adagrams.js [hunk 2]
```diff
@@ -1,8 +1,101 @@
...
+
+  highestScoreFrom(words){
+    const scores = words.map(this.scoreWord)
+    const highestScore = scores.reduce((hs, score) => hs < score ? score : hs, 0);
+    const tiedWords = [];
+   
+    for (let i in words){
+      if(words[i].length == 10) {
+        return {score: this.scoreWord(words[i]), word: words[i]};
+      }
```
Comment (droberts-sea) [inline src/adagrams.js:74]: This code gets the job done, but it could be simplified somewhat by doing the tie-breaking inline. That also lets you do the work in a single pass. We have an implementation of this for an [old version of the project](https://github.com/AdaGold/js-scrabble/blob/dpr/solution/scrabble.js) if you're interested.

**Comment by droberts-sea** [comment]
# JS Adagrams
## What We're Looking For



Feature | Feedback
--- | ---
**General** |
Answered comprehension questions | yes
Small commits with meaningful commit messages | yes
**Code Requirements** |
`drawLetters` method | yes
Uses appropriate data structure to store the letter distribution | yes
All tests for `drawLetters` pass | yes
`usesAvailableLetters` method | yes
All tests for `usesAvailableLetters` pass | yes
`scoreWord` method | yes
Uses appropriate data structure to store the letter scores | yes
All tests for `scoreWord` pass | yes
`highestScoreFrom` method | yes
Appropriately handles edge cases for tie-breaking logic | yes
All tests for `highestScoreFrom` pass | yes
**Overall** | Great job overall! This code is generally clean and well-organized, and it feels like the JavaScript unit is off to a solid start. Keep up the hard work!
